File: update-core/src/main/java/dev/bee/kanjianki/updatecore/AutoUpdateSettingsTogglePolicy.java
package dev.bee.kanjianki.updatecore;

public final class AutoUpdateSettingsTogglePolicy {
    public static final String ENABLED_MESSAGE = "Automatic updates turned on.";
    public static final String DISABLED_MESSAGE = "Automatic updates turned off.";

    private AutoUpdateSettingsTogglePolicy() {
    }

    public static ToggleResult toggle(boolean currentlyEnabled) {
        boolean enabled = !currentlyEnabled;
        return new ToggleResult(enabled, enabled ? ENABLED_MESSAGE : DISABLED_MESSAGE);
    }

    public record ToggleResult(boolean enabled, String message) {
    }
}

File: update-core/src/main/kotlin/dev/bee/kanjianki/updatecore/AutoUpdateSettingsTogglePolicy.kt
package dev.bee.kanjianki.updatecore

import java.util.Locale

object AutoUpdateSettingsTogglePolicy {
    const val ENABLED_MESSAGE = "Automatic updates turned on."
    const val DISABLED_MESSAGE = "Automatic updates turned off."

    private const val JAPANESE_LANGUAGE = "ja"

    @JvmStatic
    fun toggle(currentlyEnabled: Boolean): ToggleResult {
        val enabled = !currentlyEnabled
        return ToggleResult(
            enabled,
            if (enabled) localizedText(ENABLED_MESSAGE, "自動アップデートをオンにしました。") else localizedText(
                DISABLED_MESSAGE,
                "自動アップデートをオフにしました。",
            ),
        )
    }

    private fun localizedText(english: String, japanese: String): String =
        if (isJapaneseLocale()) japanese else english

    private fun isJapaneseLocale(): Boolean = Locale.getDefault().language == JAPANESE_LANGUAGE

    class ToggleResult(
        private val enabled: Boolean,
        private val message: String,
    ) {
        fun enabled(): Boolean = enabled
        fun message(): String = message
    }
}
